.emacs.d/test/lisp/dot-tests.el

```diff
@@ -0,0 +1,27 @@
+(require 'buttercup)
+
+(defun kotct/load-corresponding ()
+  (when noninteractive
+    (assert (string-match-p "-test\\.el$" load-file-name))
+    (load-file
+     (replace-regexp-in-string
+      "-test\\.el$" ".el"
+      (replace-regexp-in-string
+       (regexp-quote "/test/") "/"
+       load-file-name)))))
+
+(defun kotct/run-tests ()
+  (setf buttercup-suites nil)
+  (load "~/.emacs.d/test/lisp/behavior/completion-c-test.el" nil t)
```

Comment:
Is `nil t` necessary here?  Can you extract the test running behavior out into a different function and call it a bunch of times in a master function rather than add more and more of these lines to this one function?  Or maybe do globbing to find all the tests?